### ISSUE TITLE
[QUESTION]: Same results?

### ISSUE THREAD

**Issue by spyda90** [chemprop/chemprop#546]
Hi,

I have a quick question for you: using different hyperparameter optimization file during training (that differs for number of iterations during their generation, and they are different) I have obtained identical score in the test_scores.csv files (rmse, r2).

Is that either possible or am I missing something?

Thank you in advance!

Francesco

**Comment by spyda90** [comment]
If I was not clear, my question is: can different hyperparameter optimization lead to the same (identical) trained model?

Thank you,
Francesco

**Comment by kevingreenman** [comment]
When you say different hyperparameter optimization, do you mean that you've run hyperparameter optimization multiple times with different settings, and the best model from each of these hyperparameter optimization runs had different hyperparameters, but those two models gave exactly the same predictions? 

If so, I suppose that's possible but seems very unlikely unless there's something weird going on (e.g. extremely small dataset or the predicted value is the mean of the training set regardless of input).

On the other hand, if you mean that you did hyperparameter optimization with multiple different settings, but they both arrived at the same optimal set of hyperparameters, then it's expected that the resulting metrics would be identical.

**Comment by spyda90** [comment]
Hi Kevin,

Thank you for your reply.
I double-checked and you were right.

In some cases where I ran chemprop_hyperopt using the same settings except for --num_iters (30, 100, 200) I obtained identical hyperparameters (sometimes for 30 and 100 iterations, sometimes for all 30, 100, and 200).

That was the reason why when I ran chemprop_train, actually I launched identical calculations.

Thank you!
Francesco
